doc/LogitechSpotlightHID++.md

```diff
@@ -0,0 +1,137 @@
+
+```c++
+enum class FeatureCode : uint16_t {
+  Root                 = 0x0000,
+  FeatureSet           = 0x0001,
+  FirmwareVersion      = 0x0003,
+  DeviceName           = 0x0005,
+  Reset                = 0x0020,
+  DFUControlSigned     = 0x00c2,
+  BatteryStatus        = 0x1000,
+  PresenterControl     = 0x1a00,
+  Sensor3D             = 0x1a01,
+  ReprogramControlsV4  = 0x1b04,
+  WirelessDeviceStatus = 0x1db4,
+  SwapCancelButton     = 0x2005,
+  PointerSpeed         = 0x2205,
+};
+```
+
+No single Logitech device support all feature codes, rather they support a limited range of features and corresponding feature codes. Inside the device the supported feature codes are mapped to a Index (or Feature Index). This mapping is called FeatureSet table. For any device, the Root Feature Code (0x0000) has index of 0x00. Root Feature Index (0x00) is used to d
+
+For any device, the feature index corresponding to any feature code can be obtained by using Root Feature Index (0x00) by sending signal
+{0x10, 0x01, 0x00, 0x0d, Feature_Code(2 bytes), 0x00} (here, function code is 0x00 and software identification code is 0x0d in forth byte). If the return message is not an error message, fifth byte in response is Feature index and 6th byte is Feature type (See below).
+
+Software can get the entire FeatureSet table for the device with following algorithm:
+
+  1. Get the number of features supported by device:
+
+		a. Get Feature Index corresponding to Feature Set code (0x0001).
+
+		b. If the Feature Index obtained is 0x01 then get the count by sending {0x10, 0x01, 0x01, 0x0d, 0x00, 0x00, 0x00} (3rd byte is Feature Index; function code is 0x00 and software identification code is 0x0d in forth byte). In the response, 5th byte will be the number of features supported except root feature. As stated above, Root feature always has Feature Index 0x00. Hence, total number of features supported is 1 + count obtained in response.
+
+  2. Iterate over Feature Index 1 to (number of feature supported) and send the signal (assuming Feature_Index for feature set is 0x01; third byte) {0x10, 0x01, 0x01, 0x1d, Feature_Index, 0x00, 0x00} (function code is 0x10 and software identification code is 0x0d in forth byte). The response will have HID++ Feature Code in (5 and 6 byte) as uint16_t and Feature Type in 7th byte for valid Feature Index. In Feature Type byte, if 7 and 8 bit are ON then it means Software hidden and Obsolete feature respectively. So, Software_Hidden = (Feature_Type & (1<<6)) and Obsolete_Feature = (Feature_Type & (1<<7)). Software Hidden or Obsolete feature should not be handled by any software. In case Feature Index is not valid (i.e.,feature index > number of feature supported) then 0x0200 will be in (5 and 6 byte).
```

Comment:
```suggestion
  2. Iterate over the Feature Indexes 1 to  the number of features supported and send the request (assuming Feature_Index for feature set is `0x01`; third byte) `{0x10, 0x01, 0x01, 0x1d, Feature_Index, 0x00, 0x00}` (function code is `0x10` and software identification code is `0x0d` in forth byte). The response will contain the HID++ Feature Code at byte 5 and 6 as `uint16_t` and the Feature Type at byte 7 for a valid Feature Index. In the Feature Type byte, if bits 7 is set this means _Software Hidden_, if bit 8 is set this means _Obsolete feature_. So, Software_Hidden = (Feature_Type & (1<<6)) and Obsolete_Feature = (Feature_Type & (1<<7)). Software Hidden or Obsolete features should not be handled by any application. In case the Feature Index is not valid (i.e.,feature index > number of feature supported) then `0x0200` will be in the response at byte 5 and 6.
```